chef/cookbooks/network/recipes/default.rb

```diff
@@ -209,6 +209,20 @@ def get_datapath_id_for_ovsbridge(bridge)
     ifs[bond.name]["addresses"] ||= Array.new
     ifs[bond.name]["slaves"] = Array.new
     base_ifs.each do |i|
+      # If the interface is configured with an IP address, wicked will
+      # helpfully re-add the IP when the interface is downed and then
+      # brought back up -- which is required, since interfaces need to be down
+      # to be added as slaves.
+      # To stop this happening, remove the configuration file and reload the
+      # interface.
+      slave_config = "/etc/sysconfig/network/ifcfg-#{i.name}"
+      if ::File.exist?(slave_config)
+        if ::Kernel.system("grep -q IPADDR #{slave_config}")
+          Chef::Log.info("Slave #{i.name} has a configuration file, removing")
+          ::File.unlink(slave_config)
+          ::Kernel.system("wicked ifreload #{i.name}")
+        end
+      end
```

Comment:
I've had a longer discussion with the wicked developer today. And he suggested to try a slightly different approach. This is what we came up with:

1. Change Nic::Bond.add_slave() (https://github.com/crowbar/crowbar-core/blob/master/chef/cookbooks/barclamp/libraries/nic.rb#L557) to use "wicked ifdown <slave>" instead of calling "ip link set down". This would (tempoary) remove the interface from wicked`s greedy fingers ;).
2. Remove the ```slave.up``` call from Nic::Bond.add_slave() (this is actually not needed as the interface gets up automatically if the bond is up)
3. After rewriting the config files (around here: https://github.com/crowbar/crowbar-core/blob/master/chef/cookbooks/network/recipes/default.rb#L568) trigger a ```wicked ifup <interface>``` for the changed interfaces.

For completenss: He, also mentioned that it is a bad idea in general to reconfigure stuff manually (i.e. behind wicked's back) and that it would be better to either use wicked or setup things by hand. But for achieving that, we'd need to rewrite the complete networking bits in crowbar, which is probably a bit too much currently :(